manifests/hourly.pp

```diff
@@ -38,4 +33,55 @@
       require => File["${logrotate::rules_configdir}/hourly"],
     }
   }
+
+  # Make copies of the rpm provided unit and timers
+  if $logrotate::manage_systemd_timer {
+    systemd::manage_dropin { 'hourly-service.conf':
+      ensure        => $logrotate::ensure_systemd_timer,
+      unit          => 'logrotate-hourly.service',
+      unit_entry    => {
+        'Description' => [
+          '',
+          'Extra service to run hourly logrotates only',
+        ],
+      },
+      service_entry => {
+        'ExecStart'   => ['', "/usr/sbin/logrotate ${logrotate::rules_configdir}/hourly"],
+      },
+      before        => Systemd::Unit_file['logrotate-hourly.service'],
+    }
+
+    systemd::unit_file { 'logrotate-hourly.service':
+      ensure => $logrotate::ensure_systemd_timer,
+      source => 'file:///lib/systemd/system/logrotate.service',
+      before => Systemd::Unit_file['logrotate-hourly.timer'],
+    }
+
+    systemd::manage_dropin { 'hourly-timer.conf':
+      ensure      => $logrotate::ensure_systemd_timer,
+      unit        => 'logrotate-hourly.timer',
+      unit_entry  => {
+        'Description' => [
+          '',
+          'Extra timer to run hourly logrotates only',
+        ],
```

Comment:
Hi,

I don't think hourly and daily can call the same service. They have different exec lines as they operate on different config files.

As for the hourly and daily overlapping I can't immediately see why it would matter but if we could then yes makes sense to avoid it.

Quick search around systemd does not have anything obvious to mutex a couple of units.

Only suggestions I can see are an external `flock` around the logrotate command or use of some ExecCondition.

Should check if logrotate maintains it's own lock....

Reply:
Logrotate has a `--wait-for-state-lock` which sounds like the ticket.
* Need to check there's a timeout configured.
* Such a change is probably better as it's own MR since this applies to the cron variant the same.

Reply:
If that wait lock doesn't work it could be done with like `ExecStartPre=/bin/touch /var/lock/logrotate` and `ExecStartPost=/bin/rm -f /var/lock/logrotate` then have the units conditional on that path not existing.  That wait for state lock sounds better though.  A new PR for those changes sounds fine.

Reply:
Morning

If an external  lock then flock is the thing to do however I'm fairly happy with:


```
-w, --wait, --timeout seconds
    Fail if the lock cannot be acquired within seconds. Decimal fractional values are allowed. See the -E option for the exit status used. The zero
    number of seconds is interpreted as --nonblock.
```
Is the option with logrotate as in EL9.

So `-w 43200` is 12 hours seems  plenty and then the unit would enter failed state which seems fine.

Reply:
`/usr/bin/flock` now added to mutex the two instances.